bootstrap/0036_sql_tools.sql

```diff
@@ -0,0 +1,74 @@
+
+create or replace procedure admin.validate(obj object, validation_table text, is_create boolean)
+returns text
+language sql
+execute as owner
+as
+$$
+begin
+    -- Select the validation rows. In a create, choose all. In an update, omit rows which are for create_only.
+    let validation_query text := (select tools.templatejs(
```

Comment:
in the interest of fewer calls to js I am not sure if we need this. I would maybe do smoething like:
```
let validation_query text := 'select * from identifier('internal.validate_' || ?) where iff(?, true, NOT COALESCE(obj[\'create_only\'], FALSE))'
let rs resultset := (execute immediate :validation_query using (:validation_table, :is_create));
```